ros/angel_system_nodes/angel_system_nodes/spatial_mapper.py

```diff
@@ -79,14 +80,40 @@ def __init__(self):
 
         self.poses = []
 
-        # TODO: get this from a ROS message?
-        self.image_height = 1080
-        self.image_width = 1920
+        # default image dimensions used if we fail to get the current image size
+        self.image_height = 720
+        self.image_width = 1280
+
+        # attempt to get the current image size from the object detector
+        self.image_size_client = self.create_client(QueryImageSize, 'query_image_size')
+        while not self.image_size_client.wait_for_service(timeout_sec=1.0):
+            log.info("Waiting for image size service...")
+
+        r = self.send_image_size_request()
```

Comment:
I see. We can stick with the service call in the constructor. We need that information before reliably continuing on anyway, so looping until we get it seems sensible. Maybe limiting the maximum rate we query, or timing out after a while would be nice to haves, but I don't think critical given our context: it can hard-block until the service resolves.